crates/net/discv5/src/lib.rs

```diff
@@ -299,20 +314,18 @@ impl Discv5 {
     }
 
     /// Tries to convert an [`Enr`](discv5::Enr) into the backwards compatible type [`NodeRecord`],
-    /// w.r.t. local [`IpMode`]. Tries the socket from which the ENR was sent, if socket is missing
-    /// from ENR.
+    /// w.r.t. local [`IpMode`]. Uses source socket as udp socket.
     ///
-    ///  Note: [`discv5::Discv5`] won't initiate a session with any peer with a malformed node
-    /// record, that advertises a reserved IP address on a WAN network.
+    ///  Note: [`discv5::Discv5`] won't initiate a session with any peer with an unverifiable node
+    /// record, for example one that advertises a reserved LAN IP address on a WAN network. This is
+    /// in order to prevent DoS attacks, where some malicious peers may advertise a victim's socket.
```

Comment:
with this we do though. could we just specify that in such cases we try into `socket` instead of the enr record

Reply:
hm that's not a discv5 session that would result, that would be an rlpx session